File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/49862-SharePoint 2010 101 Code Samples/SharePoint Online Creating and Deploying Sandboxed Workflow Activities/C#/SPO_SandboxedWorkflowActivity/SandboxedWorkflowActivity.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Runtime.InteropServices;
using Microsoft.SharePoint;
using Microsoft.SharePoint.UserCode;

namespace SPO_SandboxedWorkflowActivity
{
    [Guid("40DE40EF-E494-4995-822C-BF623092A1E4")]
    public class SandboxedWorkflowActivity
    {
        //The activity's main public method must return a HashTable
        public Hashtable ModifyItem(SPUserCodeWorkflowContext context)
        {
            Hashtable results = new Hashtable();
            try
            {
                //Find out about the web, the list and the item in this workflow
                using (SPSite site = new SPSite(context.SiteUrl))
                {
                    using (SPWeb web = site.OpenWeb(context.WebUrl))
                    {
                        SPList list = web.Lists[context.ListId];
                        SPListItem workflowItem = list.GetItemById(context.ItemId);
                        //Modify the body field
                        workflowItem["Body"] += "This item was modified by a sandboxed workflow";
                        results["Result"] = "Success";
                        //Save the changes
                        workflowItem.Update();
                    }
                }
            }
            catch (Exception e)
            {
                results["Result"] = "Failure";
            }
            //Return the hashtable
            return results;
        }

    }
}

File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/49862-SharePoint 2010 101 Code Samples/SharePoint Online Creating and Deploying Sandboxed Workflow Activities/VB/SPO_SandboxedWorkflowActivity/SandboxedWorkflowActivity.vb
Imports System.Runtime.InteropServices
Imports Microsoft.SharePoint.UserCode

<Guid("8FC19D83-6DF0-49D9-AB86-B11F0C603290")> _
Public Class SandboxedWorkflowActivity
    'The activity's main public method must return a HashTable
    Public Function ModifyItem(ByVal context As SPUserCodeWorkflowContext) As Hashtable
        Dim results As Hashtable = New Hashtable()
        Try
            Using site As SPSite = New SPSite(context.SiteUrl)
                Using web As SPWeb = site.OpenWeb(context.WebUrl)
                    Dim list As SPList = web.Lists(context.ListId)
                    Dim workflowItem As SPListItem = list.GetItemById(context.ItemId)
                    workflowItem("Body") += "This item was modified by a sandboxed workflow"
                    results("Result") = "Success"
                    'Save the changes
                    workflowItem.Update()
                End Using
            End Using
        Catch ex As Exception
            results("Result") = "Failure"
        End Try
        Return results
    End Function
 

End Class
